packages/core/typescript/TypeScriptFileUpdate.ts

```diff
@@ -65,64 +66,120 @@ export class TypeScriptFileUpdate {
 	 * @param linkPath Routing `path` to add
 	 * @param linkText Text of the route to add as `data.text`
 	 * @param routesVariable Name of the array variable holding routes
+	 * @param parentRoutePath __Optional__ If passed, will include the new route as a **child** of the specified route path
 	 */
-	public addRoute(filePath: string, linkPath: string, linkText: string, routesVariable = "routes") {
+	public addRoute(
+		filePath: string,
+		linkPath: string,
+		linkText: string,
+		routesVariable = DEFAULT_ROUTES_VARIABLE,
+		parentRoutePath?: string
+	) {
 		let className: string;
 		const fileSource = TsUtils.getFileSource(filePath);
 		const relativePath: string = Util.relativePath(this.targetPath, filePath, true, true);
-
+		routesVariable = routesVariable || DEFAULT_ROUTES_VARIABLE;
 		className = TsUtils.getClassName(fileSource.getChildren());
 		this.requestImport([className], relativePath);
 
 		// https://github.com/Microsoft/TypeScript/issues/14419#issuecomment-307256171
 		const transformer: ts.TransformerFactory<ts.Node> = <T extends ts.Node>(context: ts.TransformationContext) =>
 			(rootNode: T) => {
-				function visitor(node: ts.Node): ts.Node {
-					if (node.kind === ts.SyntaxKind.VariableDeclaration &&
-						(node as ts.VariableDeclaration).name.getText() === routesVariable &&
-						(node as ts.VariableDeclaration).type.getText() === "Routes") {
-						// found routes variable
-						node = ts.visitEachChild(node, visitRoutesVariable, context);
-					} else {
-						node = ts.visitEachChild(node, visitor, context);
-					}
-					return node;
-				}
-				const visitRoutesVariable = (node: ts.Node): ts.Node => {
-					if (node.kind === ts.SyntaxKind.ArrayLiteralExpression) {
-						const array = (node as ts.ArrayLiteralExpression);
-
-						const routePath = ts.createPropertyAssignment("path", ts.createLiteral(linkPath));
-						const routeComponent = ts.createPropertyAssignment("component", ts.createIdentifier(className));
-						const routeDataInner = ts.createPropertyAssignment("text", ts.createLiteral(linkText));
-						const routeData = ts.createPropertyAssignment("data", ts.createObjectLiteral([routeDataInner]));
-						const newObject = ts.createObjectLiteral([routePath, routeComponent, routeData]);
-						this.createdStringLiterals.push(linkPath, linkText);
-
-						const notFoundWildCard = "**";
-						const nodes = ts.visitNodes(array.elements, visitor);
-						const errorRouteNode = nodes.filter(element => element.getText().includes(notFoundWildCard))[0];
-						let resultNodes = null;
-
-						if (errorRouteNode) {
-							resultNodes = nodes
-								.slice(0, nodes.indexOf(errorRouteNode))
-								.concat(newObject)
-								.concat(errorRouteNode);
+				let conditionalVisitor: NodeVisitor;
+				if (parentRoutePath === undefined) {
```

Comment:
This might be a nit-pick, but there's something about those big chunks inside the if-else. This can just as easily be two visitor variables and an inline condition to pick one before creating the main visitor.